docs/manual/upgrades.md

```diff
@@ -0,0 +1,129 @@
+For FoundationDB clusters that are spanned across multiple Kubernetes clusters the operator will follow a special process which is document in the [technical design](technical_design.md#bounceprocesses).
+The subreconciler will also ensure to wait until all `fdbserver` processes are ready to be restarted to prevent cases where only a subset of processes are restarted.
+After the `kill` command the operator will initiate a new reconciliation loop to detect the new running version, this is handled in the `UpdateState` subreconciler and the version is detected based on the output of the [cluster status json](https://apple.github.io/foundationdb/mr-status.html).
+
+#### Recreation of Pods Phase
+
+--> Add info incompatbile clients
+
+
+Technically the upgrade is now done and the cluster is running on the new version but the operator has to do some cleanup.
+The cleanup includes the recreation of all Pods to ensure they use the new FoundationDB image for the desired version and the configuration will be updated to point again to the binary present in the container.
+Depending on the defined [Pod update strategy](../cluster_spec.md#podupdatestrategy) the operator will handle the recreation of the Pods differently.
+In the default setting `ReplaceTransactionSystem` the operator will recreate the storage Pods zone by zone and replaces Pods of the transaction system (log and stateless processes).
+The replacement of the transaction system has the benefit that the recoveries of the transaction system is reduced to potentially one recovery.
+Once all Pods are updated to the new image the upgrade is done and the cluster status of the FoundationDB cluster resource in Kubernetes should show that the reconciliation is done.
+
+### Known issues
+
+There are a number of known issues that can occur during an upgrade of FoundationDB running on Kubernetes.
+Most of the issues are only a challenged for version incompatible upgrades and most issues are handled automatically by the operator.
+If an issue cannot be handled by the operator we will outline a way to manually resolve the issue.
+
+#### Process restarted during staging Phase
+
+During the staging phase it can happen that a `fdbserver` process gets restarted by an external event like the OOM killer or any issue that results in the `fdbserver` crashing.
+In that case the `fdbmonitor` will restart the `fdbserver` process with the new version and for version incompatible upgrades, that means this process is not able to connect to the cluster.
+The operator will mark this Process Group with the `MissingProcesses` condition, since the process will be missing in the [cluster status json](https://apple.github.io/foundationdb/mr-status.html).
+If the process has this condition longer than the [IgnoreMissingProcessesSeconds](../cluster_spec.md#foundationdbclusterautomationoptions) setting it will be ignored in the [Restart Phase](#restart-phase) to ensure the operator can move forward with the upgrade.
```

Comment:
It appears to me this step is not needed (that is, there is no need to wait for IgnoreMissingProcessesSeconds). Because those processes have already been restarted and there is no way to get them to restart at the older version (as they have been staged), and this will only delay restarting the rest of the processes/the upgrade process without giving any additional benefit.

Reply:
I believe we decided to set this value to 30s as default to prevent cases where a process is only missing for a short time. I agree with you that we should revisit the default value here to something shorter.

Created: https://github.com/FoundationDB/fdb-kubernetes-operator/issues/1519

Reply:
The problem here could be that we are using the same condition to mark processes that are in different states. Thanks for filing that issue!